crates/biome_js_analyze/src/lint/nursery/use_array_sort_compare.rs

```diff
@@ -0,0 +1,95 @@
+    /// Require Array#sort and Array#toSorted calls to always provide a compareFunction.
+    ///
+    /// When called without a compare function, Array#sort() and Array#toSorted() converts all non-undefined array elements into strings and then compares said strings based off their UTF-16 code units [ECMA specification](https://262.ecma-international.org/9.0/#sec-sortcompare).
+    ///
+    /// The result is that elements are sorted alphabetically, regardless of their type. For example, when sorting numbers, this results in a "10 before 2" order:
+    ///
+    /// ```ts,file=example.ts,ignore
+    /// [1, 2, 3, 10, 20, 30].sort(); //→ [1, 10, 2, 20, 3, 30]
+    /// ```
+    ///
+    /// This rule reports on any call to the sort methods that do not provide a compare argument.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```ts,file=invalid.ts,expect_diagnostic
+    /// const array: any[] = [];
+    /// array.sort();
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```ts,file=valid.ts
+    /// const array: any[] = [];
+    /// array.sort((a, b) => a - b);
+    /// ```
+    ///
+    pub UseArraySortCompare {
+        version: "next",
+        name: "useArraySortCompare",
+        language: "js",
+        recommended: false,
+        sources: &[RuleSource::EslintTypeScript("require-array-sort-compare").same()],
+        domains: &[RuleDomain::Project],
+    }
+}
+
+impl Rule for UseArraySortCompare {
+    type Query = Typed<JsCallExpression>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = UseArraySortCompareOptions;
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+
+        let binding = node.callee().ok()?.omit_parentheses();
+        let callee = binding.as_js_static_member_expression()?;
+
+        let call_object = callee.object().ok()?;
+        let ty = ctx.type_of_expression(&call_object);
+        if !ty.is_array_of(|_ty| true) {
+            return None;
+        }
+
+        let call_name = callee.member().ok()?.as_js_name()?.to_trimmed_text();
+        if call_name != "sort" && call_name != "toSorted" {
+            return None;
+        }
+
+        if node.arguments().ok()?.args().len() != 0 {
+            return None;
+        }
+
+        Some(())
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                node.range(),
+                markup! {
+                    "Compare function missing."
+                },
+            )
+            .note(markup! {
+                "When called without a compare function, Array#sort() and Array#toSorted() converts all non-undefined array elements into strings and then compares said strings based off their UTF-16 code units. Make sure to add a compare function to prevent unexpected sorting."
```

Comment:
nit:
```suggestion
                "When called without a compare function, Array#sort() and Array#toSorted() converts all non-undefined array elements into strings and then compares said strings based off their UTF-16 code units. Add a compare function to prevent unexpected sorting."
```
Perhaps putting the last sentence in it's own `.note()` would be good too.